tensorboard/summary/writer/event_file_writer.py

```diff
@@ -0,0 +1,226 @@
+
+    The `EventFileWriter` class creates an event file in the specified directory,
+    and asynchronously writes Event protocol buffers to the file. The Event file
+    is encoded using the tfrecord format, which is similar to RecordIO. The instance is
+    usually created and managed by `torch.utils.tensorboard.FileWriter` in pytorch.
+    """
+
+    def __init__(self, logdir, max_queue_size=10, flush_secs=120, filename_suffix=''):
+        """Creates a `EventFileWriter` and an event file to write to.
+
+        On construction the summary writer creates a new event file in `logdir`.
+        This event file will contain `Event` protocol buffers, which are written to
+        disk via the add_event method.
+        The other arguments to the constructor control the asynchronous writes to
+        the event file:
+
+        Args:
+          logdir: A string. Directory where event file will be written.
+          max_queue_size: Integer. Size of the queue for pending events and summaries.
+          flush_secs: Number. How often, in seconds, to flush the
+            pending events and summaries to disk.
+        """
+        self._logdir = logdir
+        if not os.path.exists(logdir):
+            os.makedirs(logdir)
+        self._file_name = os.path.join(logdir, "events.out.tfevents.%010d.%s.%s.%s" %
+            (time.time(), socket.gethostname(), os.getpid(), _global_uid.get())) + filename_suffix
+        self._general_file_writer = open(self._file_name, 'wb')
+        self._async_writer = _AsyncWriter(RecordWriter(self._general_file_writer), max_queue_size, flush_secs)
+
+        # Initialize an event instance.
+        _event = event_pb2.Event(wall_time=time.time(), file_version='brain.Event:2')
+        self.add_event(_event)
+        self.flush()
+
+    def get_logdir(self):
+        """Returns the directory where event file will be written."""
+        return self._logdir
+
+    def add_event(self, event):
+        """Adds an event to the event file.
+
+        Args:
+          event: An `Event` protocol buffer.
+        """
+        if not isinstance(event, event_pb2.Event):
+            raise TypeError("Expected an event_pb2.Event proto, "
+                            " but got %s" % type(event))
+        self._async_writer.write(event.SerializeToString())
+
+    def flush(self):
+        """Flushes the event file to disk.
+
+        Call this method to make sure that all pending events have been written to
+        disk.
+        """
+        self._async_writer.flush()
+
+    def close(self):
+        """Performs a final flush of the event file to disk, stops the
+        write/flush worker and closes the file. Call this method when you do not
+        need the summary writer anymore.
+        """
+        self._async_writer.close()
+
+
+class _AsyncWriter(object):
+    '''Writes bytes to an file.'''
+
+    def __init__(self, record_writer, max_queue_size=20, flush_secs=120):
+        """Writes bytes to an file asynchronously.
```

Comment:
Nit: `Writes bytes to an file asynchronously.` -> `Writes bytes to a file asynchronously.`